pravega/pravega
# Issue 6945: Lombok breaks compilation with JDK 11.0.17

Signed-off-by: Shashwat Sharma <shashwat_sharma@dell.com>

**Change log description**  
After the OS upgraded the JDK to 11.0.17 version, the build stated failing with an observation with Lombok parsing issue that breaks compilation:
```
Exception while resolving: NODE LOCAL (class com.sun.tools.javac.tree.JCTree$JCVariableDecl) @val()
final ___Lombok_VAL_Attrib__ latestInfo = (StreamSegmentInformation)store.getStreamSegmentInfo(segmentName, TIMEOUT).join()(SimpleFileObject[/home/raul/Documents/workspace/PRAVEGA-2142/pravega/segmentstore/server/src/test/java/io/pravega/segmentstore/server/store/StreamSegmentStoreTestBase.java])
java.lang.NullPointerException
        at jdk.compiler/com.sun.tools.javac.comp.Flow$FlowAnalyzer.visitApply(Flow.java:1235)
```
Another compilation issue was observed with JDK upgrade was 
```
compiler message file broken: key=compiler.misc.msg.bug arguments=11.0.17, {1}, {2}, {3}, {4}, {5}, {6}, {7}
java.lang.NullPointerException

```

which repeatedly kept the compiler to fail which eventually turned out to be an issue in OpenJDK version. 
**Purpose of the change**  
Fixes #6945 

**What the code does**  
Replaces all lombok  ``` val ``` with actual datatype after identifying all affected area.
Handle the JDK issue by refactoring the impacted code blocks after identification

**How to verify it**  
There shouldn't be any compilation issue - 
Prior to changes compiler message for the command ``` ./gradlew clean :segmentstore:storage:impl:compileTestJava ``` was:-
``` compiler message file broken: key=compiler.misc.msg.bug arguments=11.0.17, {1}, {2}, {3}, {4}, {5}, {6}, {7} ```
After the changes:- 
```
 Build JVM Version is : 11.0.17 (Ubuntu 11.0.17+8-post-Ubuntu-1ubuntu220.04)
BUILD SUCCESSFUL

 ```
The build should pass
All tests should pass
System test should pass
